uxarray/grid/geometry.py

```diff
@@ -1574,3 +1583,88 @@ def inverse_stereographic_projection(x, y, central_lon, central_lat):
     )
 
     return lon, lat
+
+
+def point_in_polygon(polygon, point, ref_point, inclusive=True):
```

Comment:
These functions will be called thousands, if not millions of times for many operations. Can you re-write this in Numba?

Reply:
When you say re-write in Numba, could you clarify? Do you mean to wrap the function in Numba, or do you mean something more?

Reply:
I'd start with wrapping it with `njit` and then seeing what modifications need to be made to be compatible. You can use the PR that I linked above with the re-writes in Numbs as a reference.